tests/nlu/training/test_train.py

```diff
@@ -63,6 +60,19 @@ def pipelines_for_tests():
                 "KeywordIntentClassifier",
             ),
         ),
+        (
+            # Created another test pipeline because dense featurizers can't co-exist in the same pipeline together,
+            # because of the mismatch in sequence length their tokenizers break the incoming message into.
```

Comment:
what about convert?

Reply:
That works as of now, along with the existing featurizers.

Reply:
hm, but convert is `1024` while spacy is `300`?

Reply:
That's true, good point. I'll create another pipeline for it then anyway. It might break later otherwise and conceptually they can't be in the same pipeline anyways.

Reply:
@tabergma Any idea, why all the tests passed with conveRT and spacy in the same pipeline? Ideally, they should have been broken with ConveRTTokenizer added.

Reply:
You mean using ConveRTTokenizer and then SpacyFeaturizer and ConveRTFeaturizer? Depends on the test. If the number of tokens for spacy and convert is the same, the pipeline works and the test does not fail.

Reply:
@tabergma  Yes, but if we change the data the tests could break. Also, regarding @Ghostvv's comment ideally with any training data, the test `test_train_model_on_test_pipelines` should fail because it would try to concatenate features from spacy and conveRT together and that concatenation should fail.
Conceptually, pulling them out into different pipelines seems like a better idea because they can't co-exist anyways because of the dimension mismatch of feature vector?

Reply:
True. So ideally the method https://github.com/RasaHQ/rasa/blob/master/rasa/nlu/featurizers/featurizer.py#L28 should not be called at all, correct? We might simply want to update the method to throw an error.

Reply:
@tabergma @Ghostvv Thought about this again. They can co-exist with different dimension sizes for last axis. concatenation of vectors of sizes (3,1024)[convert] and (3,300)[spacy] is a valid concatenation with resultant being (3,1324) dimensional. It's the first axis that matters and it shouldn't differ. The function already checks for that and throws an error.
So pulling them out into different pipelines is still a good idea because of the constraint on the first axis.

Reply:
Yeah, but most likely they will fail on real data as the first dimension will differ. So what are you suggestion: We keep it as it is and only update the test, e.g. split into different pipelines, and add a note to the documentation that you should not use multiple dense featurizers in one pipeline?

Reply:
Yes, add it to documentation saying that you should only use multiple dense featurizers if the underlying tokenization is conceptually the same. For example - subword tokenization, etc.

Reply:
I think you can use multiple dense featurizers, you cannot use multiple dense tokenizers. spacy feturizer should work with any tokenizer?

Reply:
yes, but then the number of tokens may differ which would fail

Reply:
should we add some kind of logger statement to explain why many tokenizers will fail?